mixnet/util/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+use std::{net::SocketAddr, num::NonZeroUsize, sync::Arc};
+
+use lru::LruCache;
+use parking_lot::Mutex;
+use tokio::net::TcpStream;
+
+/// The maximum number of open files allowed for the current process.
+pub static MAX_OPEN_FILES_LIMIT: once_cell::sync::Lazy<Option<usize>> =
+    once_cell::sync::Lazy::new(|| {
+        use rustix::process::{getrlimit, Resource};
+        // minus 1 because we need to leave one file descriptor for the listener
+        getrlimit(Resource::Nofile).maximum.map(|n| n as usize)
+    });
```

Comment:
In my opinion, it's safer to not have this. Instead, it would be better to force node operators to set the config (instead of `Option`).